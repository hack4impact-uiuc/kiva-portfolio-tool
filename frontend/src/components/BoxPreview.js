import React, { Component } from 'react'

import { ContentPreview } from 'box-ui-elements'
import messages from 'box-ui-elements/i18n/en-US'
import 'box-ui-elements/dist/preview.css'
import '../styles/index.scss'

import { getAccessToken } from '../utils/ApiWrapper'

const container = document.querySelector('.container')
const language = 'en-US'
const FILE_ID_VIDEO = '308566420378'
const FILE_ID_3D = '319004423111'
const FILE_ID_TEXT = '308349801521'
const FILE_ID_EXCEL = '319011536090'
const FILE_ID_AUDIO = '308566419514'
const FILE_ID_IMAGE = '308345646235'

class BoxPreview extends Component {
  constructor(props) {
    super(props)
    this.state = {
      fileId: 419075737331,
      accessToken: null
    }
  }

  async componentDidMount() {
    const res = await getAccessToken()
    if (res) {
      this.setState({
        accessToken: res
      })
    } else {
      this.setState({
        accessToken: null
      })
    }
  }

  render() {
    return (
      <ContentPreview
        hasHeader
        fileId={this.state.fileId}
        token={'xNT24dYkz8KM4QBdUnPTLvm3mAxWbgXN'}
        //token={this.state.accessToken}
        language={language}
        messages={messages}
        collection={[
          this.state.fileId,
          419072325627,
            418980052676
        ]}
      />
    )
  }
}

export default BoxPreview
