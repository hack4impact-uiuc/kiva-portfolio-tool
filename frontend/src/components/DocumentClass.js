import React, { Component } from 'react'
import { connect } from 'react-redux'
import DocumentClassPreview from './DocumentClassPreview'
import { Button, Modal, ModalBody, ModalFooter } from 'reactstrap'
import {
  deleteDocumentClass,
  updateDocumentClass,
  getAllDocumentClasses
} from '../utils/ApiWrapper'
import { bindActionCreators } from 'redux'
import { updateDocumentClasses } from '../redux/modules/user'
import { beginLoading, endLoading } from '../redux/modules/auth'
import Dropzone from 'react-dropzone'

import '../styles/documentclasspage.css'

import edit from '../media/greyEdit.png'
import remove from '../media/remove.png'

const mapStateToProps = state => ({})

const mapDispatchToProps = dispatch => {
  return bindActionCreators(
    {
      updateDocumentClasses,
      beginLoading,
      endLoading
    },
    dispatch
  )
}

/*
Corresponds to each individual Document Class in the Admin/PM overview page
Name is represented normally
Buttons exist to view, edit, and delete
View shows the description and a Box preview
*/
export class DocumentClass extends Component {
  constructor(props) {
    super(props)

    this.state = {
      name: this.props.documentClass.name,
      description: this.props.documentClass.description,
      editModal: false, // modal that appears when editing a Document class
      deleteModal: false, // modal that appears making sure the user actually wants to delete the document class
      files: []
    }

    this.editToggle = this.editToggle.bind(this)
    this.deleteToggle = this.deleteToggle.bind(this)
    this.handleDelete = this.handleDelete.bind(this)
    this.handleSubmit = this.handleSubmit.bind(this)
  }

  editToggle() {
    this.setState({ editModal: !this.state.editModal })
  }

  deleteToggle() {
    this.setState({ deleteModal: !this.state.deleteModal })
  }

  updateName = event => {
    this.setState({ name: event.target.value })
  }

  updateDescription = event => {
    this.setState({ description: event.target.value })
  }

  async handleSubmit() {
    this.props.beginLoading()
    if (this.state.files.length == 0) {
      await updateDocumentClass(
        this.props.documentClass._id,
        this.state.name,
        this.state.description,
        null,
        null
      )
    } else {
      await updateDocumentClass(
        this.props.documentClass._id,
        this.state.name,
        this.state.description,
        this.state.files[0],
        this.state.files[0].name
      )
    }
    const document_classes = await getAllDocumentClasses()
    if (document_classes) {
      this.props.updateDocumentClasses(document_classes)
    } else {
      this.props.updateDocumentClasses([])
    }
    this.props.endLoading()
    this.editToggle()
  }

  onDrop(files) {
    this.setState({
      files
    })
  }

  async handleDelete() {
    this.props.beginLoading()
    await deleteDocumentClass(this.props.documentClass._id)
    const document_classes = await getAllDocumentClasses()
    if (document_classes) {
      this.props.updateDocumentClasses(document_classes)
    } else {
      this.props.updateDocumentClasses([])
    }
    this.props.endLoading()
    this.deleteToggle()
  }

  render() {
    return (
      <>
        <Modal isOpen={this.state.editModal} toggle={this.editToggle}>
          <ModalBody>
            <form>
              <span> Name: </span>
              <input onChange={this.updateName} value={this.state.name} />
              <br />
              <span> Description: </span>
              <textarea
                name="paragraph_text"
                cols="50"
                rows="10"
                onChange={this.updateDescription}
                value={this.state.description}
              />
              <br />
              <div className="dropPage">
                <section className="droppedBox">
                  <div className="dropZone">
                    <Dropzone onDrop={this.onDrop.bind(this)}>
                      {({ getRootProps, getInputProps }) => (
                        <section>
                          <div {...getRootProps()}>
                            <input {...getInputProps()} />
                            {this.state.files.length > 0 ? (
                              <p>File uploaded: {this.state.files[0].name}</p>
                            ) : (
                              <p>Click to Upload</p>
                            )}
                          </div>
                        </section>
                      )}
                    </Dropzone>
                  </div>
                </section>
                <hr />
              </div>
            </form>
          </ModalBody>
          <ModalFooter>
            <Button className="invalidSearchButton" onClick={this.editToggle}>
              Return
            </Button>
            <Button onClick={this.handleSubmit}>Update Document Class</Button>
          </ModalFooter>
        </Modal>
        <tr className="hoverable">
          {this.props.documentClass.name ? (
            <td data-testid="docClass">{this.props.documentClass.name}</td>
          ) : null}
          <td data-testid="interaction" className="interaction">
            <DocumentClassPreview documentClass={this.props.documentClass} />
            <Button color="transparent" onClick={this.editToggle}>
              <img className="buttonimg" src={edit} />
            </Button>
            <Button color="transparent" onClick={this.deleteToggle}>
              <img className="buttonimg" src={remove} />
            </Button>
            <Modal isOpen={this.state.deleteModal} toggle={this.deleteToggle}>
              <ModalBody>
                <p>
                  Are you sure you want to delete {this.props.documentClass.name}? This will delete
                  every document of this type for all users.
                </p>
              </ModalBody>
              <ModalFooter>
                <Button className="invalidSearchButton" onClick={this.handleDelete}>
                  Delete and return
                </Button>
                <Button className="invalidSearchButton" onClick={this.deleteToggle}>
                  Return
                </Button>
              </ModalFooter>
            </Modal>
          </td>
        </tr>
      </>
    )
  }
}

export default connect(
  mapStateToProps,
  mapDispatchToProps
)(DocumentClass)
