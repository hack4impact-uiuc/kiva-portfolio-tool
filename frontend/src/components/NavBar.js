import React, { Component } from 'react'
import {
  Button,
  DropdownToggle,
  DropdownMenu,
  DropdownItem,
  Nav,
  Navbar,
  NavbarBrand,
  NavItem,
  UncontrolledDropdown
} from 'reactstrap'
import { withRouter } from 'react-router-dom'
import Sidebar from 'react-sidebar'

import { connect } from 'react-redux'

import LanguageSelector from './LanguageSelector'
import NotificationsBar from './NotificationsBar'

import k_logo from '../media/greenK.png'
import kiva_logo from '../media/kivaPlainLogo.png'
import info_image from '../media/gray_info.png'
import sandwich_image from '../media/sandwich.png'
import { removeCookie } from '../utils/cookie'
import { getUser, getFPByEmail, getPMByEmail } from '../utils/ApiWrapper'

import '../styles/navbar.scss'

const mapStateToProps = state => ({
  isPM: state.user.isPM,
  language: state.user.language
})

const sidebarClassName = ['closed', 'opened']

/**
 * This is the Navigation Bar that exist at the top of all screens once logged in
 * It contains
 * A bell icon to access the Notifications Bar
 * A sandwich icon to access a menu of other things
 */
export class NavBar extends Component {
  constructor(props) {
    super(props)

    this.state = {
      isLoginPage: null,
      sidebarOpen: false,
      sidebarClass: sidebarClassName[0],
      role: '',
      email: '',
      wrongInfo: '',
      errorMessage: ''
    }

    this.onSetSidebarOpen = this.onSetSidebarOpen.bind(this)
  }

  onSetSidebarOpen(open) {
    this.setState({ sidebarOpen: open })

    if (this.state.sidebarClass === sidebarClassName[0]) {
      this.setState({ sidebarClass: sidebarClassName[1] })
    } else {
      this.setState({ sidebarClass: sidebarClassName[0] })
    }
  }

  logout = () => {
    removeCookie('token')
    this.props.history.push('/')
  }

  redirect = async e => {
    if (this.state.role === 'fp') {
      let fp = await getFPByEmail(this.state.email)
      this.props.history.push('/dashboard/fp/' + fp._id)
    } else {
      let pm = await getPMByEmail(this.state.email)
      this.props.history.push('/overview/' + pm._id)
    }
  }

  async componentDidMount() {
    if (
      this.props.location.pathname !== '/' &&
      this.props.location.pathname !== '/register' &&
      this.props.location.pathname !== '/forgotPassword' &&
      this.props.location.pathname !== '/login'
    ) {
      this.setState({ isLoginPage: false })
    } else {
      this.setState({ isLoginPage: true })
    }

    const result = await getUser()
    if (
      result.error != null &&
      (result.error.response.status === 400 || result.error.response.status === 500)
    ) {
      this.setState({
        wrongInfo: !this.state.wrongInfo,
        errorMessage: result.error.response.data.message
      })
      return
    }

    this.setState({
      role: result.response.data.result.userRole,
      email: result.response.data.result.email
    })
  }

  languages = {
    English: {
      manage: 'Manage documents',
      changePassword: 'Change password',
      changeSecurityQuestion: 'Change security question',
      logOut: 'Log out'
    },
    Spanish: {
      manage: 'Manage documents (Spanish)',
      changePassword: 'Change password (Spanish)',
      changeSecurityQuestion: 'Change security question (Spanish)',
      logOut: 'Log out (Spanish)'
    },
    French: {
      manage: 'Manage documents (French)',
      changePassword: 'Change password (French)',
      changeSecurityQuestion: 'Change security question (French)',
      logOut: 'Log out (French)'
    },
    Portuguese: {
      manage: 'Manage documents (Portuguese)',
      changePassword: 'Change password (Portuguese)',
      changeSecurityQuestion: 'Change security question (Portuguese)',
      logOut: 'Log out (Portuguese)'
    }
  }

  render() {
    const { isPM } = this.props
    let text = this.languages[this.props.language]
    if (!text) {
      text = this.languages['English']
    }

    return (
      <div>
        {/* Currently the sidebar requires the children prop but seems like it doesn't do too much,
          so I copied the notifications bar component into the childrens prop for now to remove the console error
        */}
        {this.state.sidebarOpen ? (
          <Sidebar
            className={this.state.sidebarClass}
            rootClassName="sidebar-root"
            sidebarClassName="sidebar-styles"
            sidebar={
              <NotificationsBar
                closeFunc={this.onSetSidebarOpen}
                inDashboard={this.props.inDashboard}
              />
            }
            children={
              <NotificationsBar
                closeFunc={this.onSetSidebarOpen}
                inDashboard={this.props.inDashboard}
              />
            }
            open={this.state.sidebarOpen}
            onSetOpen={this.onSetSidebarOpen}
            pullRight={true}
          />
        ) : (
          <Sidebar
            className={this.state.sidebarClass}
            rootClassName="sidebar-root hide"
            sidebarClassName="sidebar-styles"
            sidebar={
              <NotificationsBar
                closeFunc={this.onSetSidebarOpen}
                inDashboard={this.props.inDashboard}
              />
            }
            children={
              <NotificationsBar
                closeFunc={this.onSetSidebarOpen}
                inDashboard={this.props.inDashboard}
              />
            }
            onSetOpen={this.onSetSidebarOpen}
            pullRight={true}
          />
        )}

        <Navbar className={this.props.className} color="white" light expand="md">
          {this.state.isLoginPage && (
            <NavbarBrand href="/">
              <img className="kivalogo" src={k_logo} width="60" height="60" alt="Kiva logo" />
            </NavbarBrand>
          )}

          {this.state.isLoginPage && (
            <Nav className="ml-auto margin-right-sm">
              <NavItem>
                <LanguageSelector />
              </NavItem>
            </Nav>
          )}

          {!this.state.isLoginPage && (
            <NavbarBrand onClick={this.redirect}>
              <img className="kivalogo" src={kiva_logo} width="90" height="50" alt="Kiva logo" />
            </NavbarBrand>
          )}

          {!this.state.isLoginPage && (
            <Nav className="ml-auto" id="navbar-vertical-centered" navbar>
              <NavItem className="margin-right-sm">
                <LanguageSelector />
              </NavItem>

              <NavItem>
                <Button id="info-icon" color="clear" onClick={() => this.onSetSidebarOpen(true)}>
                  <img src={info_image} width="29" height="29" alt="Info icon" />
                </Button>
              </NavItem>

              <UncontrolledDropdown className="sandwich" nav inNavbar>
                <DropdownToggle nav caret>
                  <img src={sandwich_image} width="30" height="35" alt="Sandwich icon" />
                </DropdownToggle>
                <DropdownMenu right>
                  {isPM && (
                    <div>
                      <DropdownItem onClick={() => this.props.history.push('/documentclasses')}>
                        {text.manage}
                      </DropdownItem>
                      <DropdownItem onClick={this.redirect}>Dashboard</DropdownItem>
                    </div>
                  )}
                  <DropdownItem onClick={() => this.props.history.push('/changePassword')}>
                    {text.changePassword}
                  </DropdownItem>
                  <DropdownItem onClick={() => this.props.history.push('/changeSecurityQuestion')}>
                    {text.changeSecurityQuestion}
                  </DropdownItem>
                  <DropdownItem onClick={this.logout}>{text.logOut}</DropdownItem>
                </DropdownMenu>
              </UncontrolledDropdown>
            </Nav>
          )}
        </Navbar>
      </div>
    )
  }
}

export default connect(mapStateToProps)(withRouter(NavBar))
