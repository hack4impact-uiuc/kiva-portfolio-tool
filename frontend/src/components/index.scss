body, html {
    height: 100%;
    width: 100%;
    padding: 0;
    border: 0 none;
    margin: 0;
}

.container {
    height: 100%;
    width: 100%;
}