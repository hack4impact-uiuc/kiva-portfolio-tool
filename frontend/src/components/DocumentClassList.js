import React, { Component } from 'react'
import { Table } from 'reactstrap'
import DocumentClass from './DocumentClass'
import '../styles/documentlist.css'

class DocumentClassList extends Component {
  render() {
    return (
      <Table>
        <tbody>
          <tr>
            <th className="theader-centered">Document Class Name</th>
            <th />
          </tr>
          {this.props.documentClasses
            ? this.props.documentClassess.map(documentClass => (
                <DocumentClass documentClass={documentClass} />
              ))
            : null}
        </tbody>
      </Table>
    )
  }
}

export default DocumentClassList
