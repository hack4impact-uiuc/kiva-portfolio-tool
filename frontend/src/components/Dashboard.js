import React, { Component } from 'react'
import { Container, Row, Col, Button } from 'reactstrap'

import { bindActionCreators } from 'redux'
import { connect } from 'react-redux'
import {
  updateDocuments,
  updateMessages,
  updateInstructions,
  setUserType,
  beginLoading,
  endLoading
} from '../redux/modules/user'

import WithAuth from './auth/WithAuth'
import DocumentList from './DocumentList'
import NavBar from './NavBar'

import {
  getDocumentsByUser,
  getMessagesByFP,
  updateFieldPartnerStatus,
  getFPByID
} from '../utils/ApiWrapper'

import add from '../media/add.png'

const mapStateToProps = state => ({
  isPM: state.user.isPM,
  documents: state.user.documents,
  messages: state.user.messages,
  information: state.user.information
})

const mapDispatchToProps = dispatch => {
  return bindActionCreators(
    {
      updateDocuments,
      updateMessages,
      setUserType,
      beginLoading,
      endLoading,
      updateInstructions
    },
    dispatch
  )
}
export class Dashboard extends Component {
  constructor(props) {
    super(props)

    this.state = {
      fp_statuses: ['Missing', 'Rejected', 'Pending', 'Approved'],
      pm_statuses: ['Pending', 'Missing', 'Rejected', 'Approved'],
      pm_id: null
    }

    this.handleFinish = this.handleFinish.bind(this)
  }

  async componentDidMount() {
    /**
     * Contains all documents received from backend
     */

    this.props.beginLoading()
    let documentsReceived = []

    // temporary - REMOVE after auth integration
    documentsReceived = await getDocumentsByUser(this.props.match.params.id)
    this.props.setUserType(this.props.match.params.user === 'pm')

    /**
     * Contains all messages received from backend
     */
    const messagesReceived = await getMessagesByFP(
      this.props.match.params.id,
      this.props.match.params.user === 'fp'
    )

    /**
     * Contains all information received from backend
     */
    const fp = await getFPByID(this.props.match.params.id)
    const instructionsReceived = fp.instructions
    this.setState({ pm_id: fp.pm_id })

    if (documentsReceived) {
      this.props.updateDocuments(documentsReceived)
    } else {
      this.props.updateDocuments([])
    }

    if (messagesReceived) {
      this.props.updateMessages(messagesReceived)
    } else {
      this.props.updateMessages([])
    }

    if (instructionsReceived) {
      this.props.updateInstructions(instructionsReceived)
    } else {
      this.props.updateInstructions('')
    }
    this.props.endLoading()
  }

  /**
   * When a Field Partner has finished the process, this method is called to move their status to 'Complete'
   */
  async handleFinish() {
    this.props.beginLoading()
    await updateFieldPartnerStatus(this.props.match.params.id, 'Complete')
    this.props.history.push('/overview/' + this.state.pm_id)
    this.props.endLoading()
  }

  pStyle = {
    margin: 'auto'
  }

  render() {
    return (
      <div className="background-rectangles maxheight">
        <NavBar inDashboard />
        <Container id="dashboard-container">
          {this.props.isPM ? (
            <Row>
              <Col className="text-centered" md="12">
                <Button
                  className="add-doc-text"
                  color="transparent"
                  onClick={() => this.props.history.push('/setup/' + this.props.match.params.id)}
                >
                  <img className="addImg" src={add} alt="Add icon" />
                  <span className="add-doc-text">Update requirements/instructions</span>
                </Button>
              </Col>
              <Col className="text-centered" md="12">
                <Button color="success" onClick={this.handleFinish}>
                  Finish Process
                </Button>
              </Col>
            </Row>
          ) : null}
          <Row>
            {this.props.documents
              ? this.props.isPM
                ? this.state.pm_statuses.map(key => {
                    return (
                      <Col sm="12" md="6" className="dashboard-width-override">
                        <DocumentList documents={this.props.documents[key]} status={key} />
                      </Col>
                    )
                  })
                : this.state.fp_statuses.map(key => {
                    return (
                      <Col sm="12" md="6" className="dashboard-width-override">
                        <DocumentList documents={this.props.documents[key]} status={key} />
                      </Col>
                    )
                  })
              : null}
          </Row>
        </Container>
      </div>
    )
  }
}
export default connect(
  mapStateToProps,
  mapDispatchToProps
)(WithAuth(Dashboard))
