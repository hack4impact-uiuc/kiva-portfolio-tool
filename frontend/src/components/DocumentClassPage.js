import React, { Component } from 'react'
import { Button, Modal, ModalBody, ModalFooter, Table, Input } from 'reactstrap'
import Dropzone from 'react-dropzone'

import { bindActionCreators } from 'redux'
import { connect } from 'react-redux'
import { updateDocumentClasses, beginLoading, endLoading } from '../redux/modules/user'

import WithAuth from './auth/WithAuth'
import DocumentClass from './DocumentClass'
import NavBar from './NavBar'

import { getAllDocumentClasses, createDocumentClass } from '../utils/ApiWrapper'

import add from '../media/add.png'

import '../styles/variables.scss'
import '../styles/documentclasspage.scss'

const mapStateToProps = state => ({
  documentClasses: state.user.documentClasses,
  language: state.user.language
})

const mapDispatchToProps = dispatch => {
  return bindActionCreators(
    {
      updateDocumentClasses,
      beginLoading,
      endLoading
    },
    dispatch
  )
}

/*
A page accessible by admins and PMs with an overview of all Document Classes
Functionality: Allows creation of new Document Classes and viewing/editing/deletion of existing ones through the DocumentClass component
*/
export class DocumentClassPage extends Component {
  constructor(props) {
    super(props)
    this.state = {
      addModal: false, // modal that appears after clicking 'Add New Document Class'
      name: '',
      description: '',
      files: []
    }
    this.toggle = this.toggle.bind(this)
    this.handleSubmit = this.handleSubmit.bind(this)
  }

  /**
   * Gets all document classes once component is ready
   */
  async componentDidMount() {
    const document_classes = await getAllDocumentClasses()
    if (document_classes) {
      this.props.updateDocumentClasses(document_classes)
    } else {
      this.props.updateDocumentClasses([])
    }
  }

  /**
   * Opens/Closes the addModal
   */
  toggle() {
    this.setState({ addModal: !this.state.addModal })
  }

  /**
   * updates name in state
   */
  updateName = event => {
    this.setState({ name: event.target.value })
  }

  /**
   * updates description in state
   */
  updateDescription = event => {
    this.setState({ description: event.target.value })
  }

  /**
   * updates document class and closes add modal
   */
  async handleSubmit() {
    this.props.beginLoading()
    await createDocumentClass(
      this.state.name,
      this.state.description,
      this.state.files[0],
      this.state.files[0].name
    )
    const document_classes = await getAllDocumentClasses()
    if (document_classes) {
      this.props.updateDocumentClasses(document_classes)
    } else {
      this.props.updateDocumentClasses([])
    }
    this.props.endLoading()
    this.toggle()
  }

  onDrop(files) {
    this.setState({
      files
    })
  }

  languages = {
    English: {
      name: 'Name:',
      description: 'Description:',
      fileUploaded: 'File uploaded: ',
      click: 'Click to upload',
      return: 'Return',
      close: 'Close',
      create: 'Create new document class',
      edit: 'Edit document classes',
      add: 'Add new document class',
      docClassName: 'Document class name'
    },
    Spanish: {
      name: 'Name: (Spanish)',
      description: 'Description: (Spanish)',
      fileUploaded: 'File uploaded:  (Spanish)',
      click: 'Click to upload (Spanish)',
      return: 'Return (Spanish)',
      close: 'Close (Spanish)',
      create: 'Create new document class (Spanish)',
      edit: 'Edit document classes (Spanish)',
      add: 'Add new document class (Spanish)',
      docClassName: 'Document class name (Spanish)'
    },
    French: {
      name: 'Name: (French)',
      description: 'Description: (French)',
      fileUploaded: 'File uploaded:  (French)',
      click: 'Click to upload (French)',
      return: 'Return (French)',
      close: 'Close (French)',
      create: 'Create new document class (French)',
      edit: 'Edit document classes (French)',
      add: 'Add new document class (French)',
      docClassName: 'Document class name (French)'
    },
    Portuguese: {
      name: 'Name: (Portuguese)',
      description: 'Description: (Portuguese)',
      fileUploaded: 'File uploaded:  (Portuguese)',
      click: 'Click to upload (Portuguese)',
      return: 'Return (Portuguese)',
      close: 'Close (Portuguese)',
      create: 'Create new document class (Portuguese)',
      edit: 'Edit document classes (Portuguese)',
      add: 'Add new document class (Portuguese)',
      docClassName: 'Document class name (Portuguese)'
    }
  }

  render() {
    let text = this.languages[this.props.language]
    if (!text) {
      text = this.languages['English']
    }

    return (
      <div className="background-rectangles maxheight">
        <NavBar />
        <Modal isOpen={this.state.addModal} toggle={this.toggle}>
          <ModalBody>
            <form>
              <span>{text.name}</span>
              <Input type="textarea" className="textarea-input" onChange={this.updateName} />
              <br />
              <span>{text.description}</span>
              <Input
                type="textarea"
                className="textarea-input"
                style={{ height: '200px' }}
                onChange={this.updateDescription}
              />
              <br />
              <div className="dropPage">
                <section className="droppedBox">
                  <div className="dropZone">
                    <Dropzone onDrop={this.onDrop.bind(this)}>
                      {({ getRootProps, getInputProps }) => (
                        <section>
                          <div {...getRootProps()}>
                            <input {...getInputProps()} />
                            {this.state.files.length > 0 ? (
                              <p>{text.fileUploaded + this.state.files[0].name}</p>
                            ) : (
                              <p>{text.click}</p>
                            )}
                          </div>
                        </section>
                      )}
                    </Dropzone>
                  </div>
                </section>
                <hr />
              </div>
            </form>
          </ModalBody>
          <ModalFooter>
            <Button className="invalidSearchButton" onClick={this.toggle}>
              {text.close}
            </Button>
            <Button
              disabled={this.state.files.length === 0}
              color="success"
              onClick={this.handleSubmit}
            >
              {text.create}
            </Button>
          </ModalFooter>
        </Modal>
        <div className="edit-banner">
          <h1 className="h1">{text.edit}</h1>
          <Button
            className="add-doc-text"
            id="add-new-class"
            color="transparent"
            onClick={this.toggle}
          >
            <img className="addImg" src={add} alt="Add icon" />
            <span className="add-doc-text">{text.add}</span>
          </Button>
        </div>

        <div className="doc-table">
          <Table>
            <tbody>
              <tr id="doc-table-top-row">
                <th className="theader-centered">{text.docClassName}</th>
                <th />
              </tr>
              {this.props.documentClasses
                ? this.props.documentClasses.map(documentClass => (
                    <DocumentClass documentClass={documentClass} />
                  ))
                : null}
            </tbody>
          </Table>
        </div>

        <div className="returnBtnContainer text-centered padding-bottom-sm">
          <Button className="returnButton" onClick={this.props.history.goBack}>
            {text.return}
          </Button>
        </div>
      </div>
    )
  }
}

export default connect(
  mapStateToProps,
  mapDispatchToProps
)(WithAuth(DocumentClassPage))
