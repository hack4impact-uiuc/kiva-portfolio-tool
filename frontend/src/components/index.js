import Dashboard from './Dashboard'
import DocumentPreview from './DocumentPreview'

import SelectDocumentsPage from './SelectDocuments'
import BoxPreview from './DocumentPreview'
import PMMainPage from './PMMainPage'
import DocumentClassPage from './DocumentClassPage.js'

import Load from './Load'

import ErrorMessage from './auth/ErrorMessage'
import ForgotPassword from './auth/ForgotPassword'
import LogIn from './auth/LogIn'
import Register from './auth/Register'
import WrongPage from './auth/WrongPage'

export {
  Dashboard,
  LogIn,
  BoxPreview,
  Register,
  SelectDocumentsPage,
  PMMainPage,
  ForgotPassword,
  DocumentClassPage,
  DocumentPreview,
  Load,
  WrongPage,
  ErrorMessage
}
