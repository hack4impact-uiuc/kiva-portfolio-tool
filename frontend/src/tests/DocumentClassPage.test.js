import React from 'react'
import { DocumentClassPage } from '../components/DocumentClassPage'
import { NavBar } from '../components/NavBar'
import { Button, Modal, ModalBody, ModalFooter, Table, Label, Media } from 'reactstrap'
import Adapter from 'enzyme-adapter-react-16'
import { shallowToJson, mountToJson } from 'enzyme-to-json'
import { configure, shallow, mount } from 'enzyme'
import chai from 'chai'
global.jestExpect = global.expect
global.expect = chai.expect
import sinon from 'sinon'

beforeAll(() => {
  configure({ adapter: new Adapter() })
})

test('DocumentClassPage renders correctly', () => {
  const component = shallow(<DocumentClassPage />)

  let tree = shallowToJson(component)
  jestExpect(tree).toMatchSnapshot()
})

describe('<DocumentClassPage />', () => {
  it('renders All major components', () => {
    const wrapper = shallow(<DocumentClassPage />)
    expect(wrapper.find(Modal)).to.have.lengthOf(1)
    expect(wrapper.find(Button)).to.have.lengthOf(3)
    expect(wrapper.find(Table)).to.have.lengthOf(1)
  })

  it('renders css in modal', () => {
    const wrapper = shallow(<DocumentClassPage />)
    expect(wrapper.find('.dropPage')).to.have.lengthOf(1)
    expect(wrapper.find('.droppedBox')).to.have.lengthOf(1)
    expect(wrapper.find('.dropZone')).to.have.lengthOf(1)
    expect(wrapper.find('.invalidSearchButton')).to.have.lengthOf(1)
  })

  it('renders css in button', () => {
    const wrapper = shallow(<DocumentClassPage />)
    expect(wrapper.find('.edit-banner')).to.have.lengthOf(1)
    expect(wrapper.find('.h1')).to.have.lengthOf(1)
    expect(wrapper.find('.add-doc-text')).to.have.lengthOf(2)
    expect(wrapper.find('.addImg')).to.have.lengthOf(1)
  })

  it('renders css in table ', () => {
    const wrapper = shallow(<DocumentClassPage />)
    expect(wrapper.find('.doc-table')).to.have.lengthOf(1)
    expect(wrapper.find('.theader-centered')).to.have.lengthOf(1)
  })
})
