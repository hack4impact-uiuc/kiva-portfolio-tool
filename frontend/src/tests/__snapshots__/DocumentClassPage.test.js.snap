// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`DocumentClassPage renders correctly 1`] = `
<Fragment>
  <Connect(withRouter(NavBar)) />
  <Modal
    autoFocus={true}
    backdrop={true}
    backdropTransition={
      Object {
        "mountOnEnter": true,
        "timeout": 150,
      }
    }
    centered={false}
    fade={true}
    isOpen={false}
    keyboard={true}
    modalTransition={
      Object {
        "timeout": 300,
      }
    }
    onClosed={[Function]}
    onOpened={[Function]}
    role="dialog"
    toggle={[Function]}
    zIndex={1050}
  >
    <ModalBody
      tag="div"
    >
      <form>
        <span>
           Name: 
        </span>
        <input
          onChange={[Function]}
        />
        <br />
        <span>
           Description: 
        </span>
        <textarea
          cols="50"
          name="paragraph_text"
          onChange={[Function]}
          rows="10"
        />
        <br />
        <div
          className="dropPage"
        >
          <section
            className="droppedBox"
          >
            <div
              className="dropZone"
            >
              <Dropzone
                onDrop={[Function]}
              >
                <Component />
              </Dropzone>
            </div>
          </section>
          <hr />
        </div>
      </form>
    </ModalBody>
    <ModalFooter
      tag="div"
    >
      <Button
        className="invalidSearchButton"
        color="secondary"
        onClick={[Function]}
        tag="button"
      >
        Return
      </Button>
      <Button
        color="secondary"
        disabled={true}
        onClick={[Function]}
        tag="button"
      >
        Create Document Class
      </Button>
    </ModalFooter>
  </Modal>
  <div
    className="edit-banner"
  >
    <h1
      className="h1"
    >
      Edit Document Classes
    </h1>
    <Button
      className="add-doc-text"
      color="transparent"
      id="add-new-class"
      onClick={[Function]}
      tag="button"
    >
      <img
        className="addImg"
        src="add.png"
      />
      <span
        className="add-doc-text"
      >
        Add New Document Class
      </span>
    </Button>
  </div>
  <div
    className="doc-table"
  >
    <Table
      responsiveTag="div"
      tag="table"
    >
      <tbody>
        <tr
          id="doc-table-top-row"
        >
          <th
            className="theader-centered"
          >
            Document Class Name
          </th>
          <th />
        </tr>
      </tbody>
    </Table>
  </div>
</Fragment>
`;
